dwitter/views_v2.py

```diff
@@ -150,10 +161,24 @@ def list(self, request):
 
         return super().list(request)
 
+    def destroy(self, request, *args, **kwargs):
+        if not request.user.is_authenticated:
+            raise PermissionDenied()
+
+        dweet = self.get_object()
+
+        # Only allow deletion of own dweets
+        # unless the deleter is a mod
+        if request.user == dweet.author or request.user.is_staff:
+            dweet.delete()
```

Comment:
.delete() on an instance is built-in functionality from django I think. Will this actually delete from the db instead of setting the deleted flag to True?

Reply:
Soft delete was implemented by overriding the delete() function. 
I've tested the new [deleted] views locally, and I can't see how that would work if this was a full delete.
https://github.com/lionleaf/dwitter/blob/a70a82c4854dd192dd9e910df515d1082690e85e/dwitter/models.py#L50